### ISSUE TITLE
Bug: [rule no-empty-interface not work true] It should allow me to extend an interface from another interface with `omit`

### ISSUE THREAD

**Issue by nqhd3v** [typescript-eslint/typescript-eslint#7159]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=script&code=JYOwLgpgTgZghgYwgAgJIGEoTpAgghAewFdxkBvAWAChlliBnaEOAWwgC5kGwpQBzANw06ABzgMGAd0JQAJlx58QQkciiEANhAD8i3gOG1kc4A1Ga4ATwD6LdvuWrjEVnGCbHhmgF8aNCAAPUVkwZFBIWEQUVHwiUjAAJQhzQhAmNSDIEDkGZAB5VmAwAB4MLBwIOJJwABpkAHJxSRl5BoA+CmQfIA&eslintrc=N4KABGBEAOCGBOBnApvSAuKABALgT2mUQGN4BLaHAWiIBsyA7HAejiVUgBpwo2V4A8pTIB7BogxhQECDHgiAVsmI5JkHCTEAzMgHMAdAsRiuPWcYCu8YsgAqBZGoC2IgCYXajngF9us6LQWuowSmADakLgOJOSUNIj0TMx0jNQBQYyQPAC6flDIAB44yAyuoWBhZryBwQzoUYQxFNQpSfDKIk5OJa7IrqYy1Rl10O04OGSozO3End2lfVkQuTyQ8iKqmDjwFsh5kCUAbpLSMpAMbo5bO3tVkEqIm2Dbuz77ehftAAqw46jikgi+la22IhgkK1kO085VOsgaRFIzXiiRYhQCZGIZGoWgsDBUogYVDGViJ+EIahEWi0AzOCKacRByQKGKx1Bc7k8VAARiI8a4EHgqOSiJTqbT4SKGS0EqlmBcaCz6GyqLAGHgxTS8pLokjGbKkgrkE5KELUqgtLAbJqlmBvCB7UA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEk8sAJgK4YVRzz1EBWZANOBqQG6aKgCGmcQWJs8mJugE0omdPQBu6AMIJ46InjKV5i5UPXI8AEVGwpeewFl0ecQ2ni5CpYNroAD0QsZH1DcVRbOAd7YzBTXxVICNRYAHcAZQBPQwALV2RoWwAzcRY8AEkkeyNvMz8oN0wAc1c2SHQSACYABgBGAHZIc1oSWCZMaHQncUQ6xKFxm2RMdoA6AHoGZBI8YYbIACNxEnQAVUxUdY39pOQiaDUwtyuTHxGoEgBrZEQAGWQh20+WgX3mH0gu0whTwADkWKhgaIvnFSqhThCiLAqohUIUrABBIhZNhojEHKEwgBC9wY2hSBMQuJJlDJC1oxXsU10pB2eAMeHpJHuzQqRAAYmh0LDxPpURFyUkseKUnhcphxs1ckLOmKMmkrNBcqSFehBABfEDmoA

### Repro Code

```typescript
export interface ICreateAccount {
  username: string;
  password: string;
  role: Role;
  display_name: string;
  email: string;
  updateID() {}
}

export interface IAccountResponse
  extends Omit<ICreateAccount, 'password' | 'updateID'> {}
```

### ESLint Config

```javascript
// Default config when I initial source code with Nestjs (`nest new app`)
module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
    tsconfigRootDir: __dirname,
    sourceType: 'module',
  },
  plugins: ['@typescript-eslint/eslint-plugin'],
  extends: [
    'plugin:@typescript-eslint/recommended',
    'plugin:prettier/recommended',
  ],
  root: true,
  env: {
    node: true,
    jest: true,
  },
  ignorePatterns: ['.eslintrc.js'],
  rules: {
    '@typescript-eslint/interface-name-prefix': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

In this case, I am trying to create a new interface, which extends from another interface, but it doesn't include some fields. It is different with I create another interface, which is the same original interface.

```typescript
// incorrect: an interface with only one supertype (Bar === Foo)
interface Bar extends Foo {}
```

### Actual Result

It throws an ESlint error:

![image](https://github.com/typescript-eslint/typescript-eslint/assets/106160187/0c25d470-5390-4c52-bee2-f4f0c8767b90)

**Comment by Josh-Cena** [comment]
Why not `type IAccountResponse = Omit<ICreateAccount, "password" | "updateID">`?

**Comment by nqhd3v** [comment]
I know that can work, but I am trying to use `interface`.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-empty-interface/#options

There's an option for you already.

**Comment by Josh-Cena** [comment]
I believe they wanted to ban `interface Bar extends Foo`, but allow complex `extends` (as a replacement for type aliases)...

**Comment by bradzacher** [comment]
If that was the desire - super niche usecase that we shan't be supporting...
Disable comment is the way to go to allow you to document why you're circumventing the rule for this specific usecase.
